### ISSUE TITLE
Automatically connect network

### ISSUE THREAD

**Issue by mrmarbury** [CenturyLinkCloud/chef-provisioning-vsphere#19]
Hi,

I have the problem that the network card is not being connected automatically on kitchen setup. Is there an Option for this? 

Cheers,
Stefan

**Comment by mwrock** [comment]
The driver will do this by default. If the nic is not connected, that usually means there is something wrong with the network assigned in the config. Have you checked the vmware client to see if its network matches up with the one that you assigned it in the config?

**Comment by mrmarbury** [comment]
the network_name attribute is set to the value of the network name field in the client. When I configure a different network_name that is not existent I get an error message anyway. So I guess this is correct then?

**Comment by mwrock** [comment]
Is the same network already assigned to the template? I need to check but it might just do nothing if the template being cloned has that network attached.

Also, if you manually mark it as connected in the vmware client, does the nic work?

**Comment by mrmarbury** [comment]
The same network is already attached in the template. The following doesn't matter:
- the base is a vm or a template
- I comment out the network_name attribute in the kitchen config
- the test vm is a linked clone (or not)

When I remove the nic of the template I get an error. And since I can only select one network, this is no option.

When I manually connect the nic, everything wworks and I can converge, destroy, what not. So it is really only that the nic is not attached

**Comment by mwrock** [comment]
I am trying to reproduce this scenario without any luck. Can you run test-kitchen with `-l debug` and include the config spec in the debug output. In that output, you should se something like:

```
    deviceChange: [VirtualDeviceConfigSpec(
       device: VirtualVmxnet3(
         backing: VirtualEthernetCardNetworkBackingInfo(
           deviceName: "network_name"
         ),
         connectable: VirtualDeviceConnectInfo(
           allowGuestControl: true,
           connected: true,
           startConnected: true
         ),
         deviceInfo: Description(
           label: "Ethernet 1",
           summary: "network_name"
         ),
         key: 4000
       ),
       operation: #<RbVmomi::VIM::VirtualDeviceConfigSpecOperation:0x00000005ba3b28
        @value="edit">
     )],
```
That would help to further debug this issue.

**Comment by mrmarbury** [comment]
Oh well, after some fiddeling around I removed the whole customization spec in a "catching at a straw"-attempt. Now it works ^^

I don't need any customization at all so the spec was just this stub anyway:

```
customization_spec:
  domain: <my domain>
    ipsettings:
    dnsServerList:
```

and I removed the network_name attribute which lead to a
```Error: Fault.NicSettingMismatch.Summary```

My template is configured to use DHCP and I can only select one network anyway so this is set by default.

Thank you for your help :) and maybe this helps anybody with a similar problem.

Cheers,
Stefan
